### PR TITLE
Add verbs to Open/Close Openable containers, and add optional seals

### PR DESCRIPTION
**Pull request by Tayrtahn**
## About the PR

Bottles, cans, and anything else that can be opened can now be opened using the verbs/context menu. Some opened items can also now be closed using the same menu, while previously there was no way to close them. Items like soda cans and condiment packets cannot be closed again after opening.

Additionally, items can optionally have a seal, which will be broken when the item is opened and cannot be resealed. This is used on most liquor bottles, drink cartons, and water/soda bottles, which all have fairly evident tamper seals.

## Why / Balance

While working on #24600 (which will enhance this PR) I found these options were missing. The seals are kinda fluff, but they make sense and could be used to catch someone attempting to poison a bottle.

## Technical details

Verbs and Closing functionality expand OpenableComponent and OpenableSystem. Seals are implemented by a new SealableComponent and System. Openables now raise OpenableOpenedEvent and OpenableClosedEvent when being opened/closed.

Open and Close verb labels and sealed/unsealed descriptions can be customized to support non-drink Openables (ex: "Unwrap/Wrap", "The foil is torn"). Closeable is false by default to minimize impact on existing content.

SealableComponent's state is exposed for visualizing in Appearance by the SealableVisuals enum. Nothing uses this right now, but it's there if anyone wants to make seals visible on something in the future.

## Media

Open/Close Verbs:
Screenshot 2024-01-31 at 2 24 16 PM
Screenshot 2024-01-31 at 2 24 36 PM
Seal examine text:
Broken seal
Intact seal

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Drink bottles can now be opened and closed from the verbs menu, and some have been given tamper-evident seals.

### CODE REVIEW

File: Content.Server/Nutrition/Components/OpenableComponent.cs [hunk 1]
```diff
@@ -38,9 +38,63 @@ public sealed partial class OpenableComponent : Component
     [DataField, ViewVariables(VVAccess.ReadWrite)]
     public LocId ClosedPopup = "drink-component-try-use-drink-not-open";
 
+    /// <summary>
+    /// Text to show in the verb menu for the "Open" action.
+    /// You may want to change this for non-drinks, i.e. "Peel", "Unwrap"
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```
Comment (deltanedas) [inline Content.Server/Nutrition/Components/OpenableComponent.cs:45]: for all of these just [DataField] its automatically readwrite

File: Content.Server/Nutrition/Components/OpenableComponent.cs [hunk 2]
```diff
@@ -38,9 +38,63 @@ public sealed partial class OpenableComponent : Component
...
     /// Sound played when opening.
     /// </summary>
     [DataField]
     public SoundSpecifier Sound = new SoundCollectionSpecifier("canOpenSounds");
+
+    /// <summary>
+    /// Can this item be closed again after opening?
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public bool Closeable = false;
```
Comment (deltanedas) [inline Content.Server/Nutrition/Components/OpenableComponent.cs:65]: `= false` isnt needed thats the default

File: Content.Server/Nutrition/Components/OpenableComponent.cs [hunk 3]
```diff
@@ -38,9 +38,63 @@ public sealed partial class OpenableComponent : Component
...
+    public SoundSpecifier? CloseSound;
+
+    /// <summary>
+    /// If true, opening and reclosing this item will leave
+    /// evidence that it has been opened, shown in the
+    /// examined text. If false, no special text will be
+    /// displayed.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public bool Sealable = false;
```
Comment (deltanedas) [inline Content.Server/Nutrition/Components/OpenableComponent.cs:80]: above

it also may be worth putting this in its own component or something

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Server/Nutrition/Components/OpenableComponent.cs (hunk 3 above)
Comment (Tayrtahn) [reply Content.Server/Nutrition/Components/OpenableComponent.cs:80]: Yeah, it probably would be cleaner to do a separate SealableComponent and have Openable raise an event on open. I'll work on that.

File: Content.Server/Nutrition/EntitySystems/SealableSystem.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    private void OnExamined(EntityUid uid, SealableComponent comp, ExaminedEvent args)
+    {
+        if (!args.IsInDetailsRange)
+            return;
+
+        string sealedText;
+        if (comp.Sealed)
+            sealedText = Loc.GetString(comp.ExamineTextSealed);
+        else
+            sealedText = Loc.GetString(comp.ExamineTextUnsealed);
```
[the remark this answers is not in the document]
Comment (Tayrtahn) [reply Content.Server/Nutrition/EntitySystems/SealableSystem.cs:25]: Ah, good call

**Comment by Tayrtahn** [comment]
This should be ready to go now.

File: Content.Server/Nutrition/Components/SealableComponent.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    /// </summary>
+    [DataField]
+    public LocId ExamineTextSealed = "drink-component-on-examine-is-sealed";
+
+    /// <summary>
+    /// Text shown when examining and the item's seal has been broken.
+    /// </summary>
+    [DataField]
+    public LocId ExamineTextUnsealed = "drink-component-on-examine-is-unsealed";
+}
```
Comment (metalgearsloth) [inline Content.Server/Nutrition/Components/SealableComponent.cs:30]: Move to shared + network.

File: Content.Server/Nutrition/EntitySystems/OpenableSystem.cs [hunk 6]
```diff
@@ -139,4 +175,33 @@ public bool TryOpen(EntityUid uid, OpenableComponent? comp = null)
...
+public sealed class OpenableOpenedEvent : EventArgs
+{
+}
+
+/// <summary>
+/// Raised after an Openable is closed.
+/// </summary>
+public sealed class OpenableClosedEvent : EventArgs
+{
 }
```
Comment (metalgearsloth) [inline Content.Server/Nutrition/EntitySystems/OpenableSystem.cs:207]: Make these record struct events raised by ref.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Audio asset is missing a licence.

**Comment by Tayrtahn** [comment]
The audio is just the first part of bottle_open1 run through a high pass filter. bottle_open1 doesn't seem to have any licence info, so I'm not sure what to put. 🤷

I'm not really happy with it though, so I'll just replace it.

**Pull request merged**